packages/flutter/lib/src/material/carousel.dart

```diff
@@ -579,6 +579,10 @@ class _RenderSliverFixedExtentCarousel extends RenderSliverFixedExtentBoxAdaptor
 
   // This implements the [itemExtentBuilder] callback.
   double _buildItemExtent(int index, SliverLayoutDimensions currentLayoutDimensions) {
+    if (maxExtent == 0.0) {
```

Comment:
The `UnsupportedOperationError`'s were all caused by divisions on the max extent, when it is zero.

This new check + the ones on line 648 & line 684 fix those issues.